pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/BufferedData.java

```diff
@@ -297,8 +297,10 @@ public long remaining() {
      * {@inheritDoc}
      */
     @Override
-    public long skip(long count) {
-        count = Math.min(count, buffer.remaining());
+    public long skip(final long count) {
+        if (count > Integer.MAX_VALUE || (int) count > buffer.remaining()) {
```

Comment:
I wonder if the cast to (int) is really needed here

Reply:
I've added it for clarity. The compiler would perform the cast anyway because buffer.remaining() is int, but it could instead expand that int to a long. So I'm just explicitly telling the compiler to down-cast the count to int for the purpose of the comparison now that I've ensured that the value fits in an integer type in the previous condition.